src/tests/encore/parametric/methodPolyClosure.enc

```diff
@@ -0,0 +1,18 @@
+read class Bar[t]
+  def bar() : unit
+    println("In Bar")
+  end
+end
+
+read class Foo
+  def foo[ty]() : () -> unit
+    fun() => (new Bar[ty]()).bar()
```

Comment:
The bug this example hits has been reported as #809 — it seems to be not caused by this PR.